src/Interpreters/Access/InterpreterCreateUserQuery.cpp

```diff
@@ -22,18 +27,22 @@ namespace
         const std::shared_ptr<ASTUserNameWithHost> & override_name,
         const std::optional<RolesOrUsersSet> & override_default_roles,
         const std::optional<SettingsProfileElements> & override_settings,
-        const std::optional<RolesOrUsersSet> & override_grantees)
+        const std::optional<RolesOrUsersSet> & override_grantees, const bool allow_plaintext_and_no_password)
     {
         if (override_name)
             user.setName(override_name->toString());
         else if (!query.new_name.empty())
             user.setName(query.new_name);
         else if (query.names->size() == 1)
             user.setName(query.names->front()->toString());
-
         if (query.auth_data)
             user.auth_data = *query.auth_data;
 
+        //User and  query IDENTIFIED WITH AUTHTYPE PLAINTEXT and NO_PASSWORD should not be allowed if allow_plaintext_and_no_password is unset.
+        if ((user.auth_data.getType() == AuthenticationType::NO_PASSWORD || query.auth_data->getType() == AuthenticationType::PLAINTEXT_PASSWORD || query.auth_data->getType() == AuthenticationType::NO_PASSWORD) &&  !allow_plaintext_and_no_password)
+            throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "InterPreterQuery: User is not allowed to ALTER/CREATE USERS with type "+ toString(user.auth_data.getType())+". Please configure User with authtype"
```

Comment:
"InterPreterQuery: " at the beginning of the message is superfluous, because all exceptions are always joined with call stack anyway